src/domain/authentication/index.types.d.ts

```diff
@@ -4,6 +4,7 @@ type AuthenticationError = import("./errors").AuthenticationError
 
 type IdentityPassword = string & { readonly brand: unique symbol }
 
+// can be either kratosUserId or deviceId subjects
```

Comment:
This comment indicates missing domain concepts.

Reply:
maybe we can add another TODO here. we should tackle this when this PR is merged I think.